From 211f3eb2690c0f45ed6ca29d21f35562f88e83c1 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:58:58 -0400
Subject: [ethernet] mlx: Convert ethernet mac address declarations to use ETH_ALEN

Message-id: <6fdd46901dd318e94f4c3932822f3baad07ec627.1410425016.git.avadai@redhat.com>
Patchwork-id: 93576
O-Subject: [Patch RHEL7.1 V1 002/197] include: Convert ethernet mac address declarations to use ETH_ALEN
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Joe Perches <joe@perches.com>

It's convenient to have ethernet mac addresses use
ETH_ALEN to be able to grep for them a bit easier and
also to ensure that the addresses are __aligned(2).

Add #include <linux/if_ether.h> as necessary.

Signed-off-by: Joe Perches <joe@perches.com>
Acked-by: Mauro Carvalho Chehab <m.chehab@samsung.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 574e2af7c0af3273836def5e66f236521bb433c9)

only included mlx parts from the upstream commit
CC: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/mlx4/device.h b/include/linux/mlx4/device.h
index 8212ab3..2bb4156 100644
--- a/include/linux/mlx4/device.h
+++ b/include/linux/mlx4/device.h
@@ -33,6 +33,7 @@
 #ifndef MLX4_DEVICE_H
 #define MLX4_DEVICE_H
 
+#include <linux/if_ether.h>
 #include <linux/pci.h>
 #include <linux/completion.h>
 #include <linux/radix-tree.h>
@@ -630,7 +631,7 @@ struct mlx4_eth_av {
  u8  dgid[16];
  u32  reserved4[2];
  __be16  vlan;
- u8  mac[6];
+ u8  mac[ETH_ALEN];
 };
 
 union mlx4_ext_av {
@@ -943,10 +944,10 @@ enum mlx4_net_trans_promisc_mode {
 };
 
 struct mlx4_spec_eth {
- u8 dst_mac[6];
- u8 dst_mac_msk[6];
- u8 src_mac[6];
- u8 src_mac_msk[6];
+ u8 dst_mac[ETH_ALEN];
+ u8 dst_mac_msk[ETH_ALEN];
+ u8 src_mac[ETH_ALEN];
+ u8 src_mac_msk[ETH_ALEN];
  u8 ether_type_enable;
  __be16 ether_type;
  __be16 vlan_id_msk;
diff --git a/include/linux/mlx4/qp.h b/include/linux/mlx4/qp.h
index 2cda9f3..59f8ba8 100644
--- a/include/linux/mlx4/qp.h
+++ b/include/linux/mlx4/qp.h
@@ -34,6 +34,7 @@
 #define MLX4_QP_H
 
 #include <linux/types.h>
+#include <linux/if_ether.h>
 
 #include <linux/mlx4/device.h>
 
@@ -147,7 +148,7 @@ struct mlx4_qp_path {
  u8   feup;
  u8   fvl_rx;
  u8   reserved4[2];
- u8   dmac[6];
+ u8   dmac[ETH_ALEN];
 };
 
 enum { /* fl */
@@ -324,7 +325,7 @@ struct mlx4_wqe_datagram_seg {
  __be32   dqpn;
  __be32   qkey;
  __be16   vlan;
- u8   mac[6];
+ u8   mac[ETH_ALEN];
 };
 
 struct mlx4_wqe_lso_seg {
-- 
1.7.1